app/src/main/java/org/mifos/mobile/ui/update_password/UpdatePasswordScreen.kt

```diff
@@ -0,0 +1,253 @@
+                        }
+                    } else {
+                        Icon(imageVector = Icons.Filled.Error, contentDescription = null)
+                    }
+                },
+                error = newPasswordError,
+                visualTransformation = if (viewNewPassword) VisualTransformation.None else PasswordVisualTransformation(),
+                keyboardType = KeyboardType.Password
+            )
+
+            MifosOutlinedTextField(
+                value = confirmPassword,
+                onValueChange = {
+                    confirmPassword = it
+                    confirmPasswordError = false
+                },
+                label = R.string.confirm_password,
+                supportingText = confirmPasswordErrorContent,
+                icon = R.drawable.ic_lock_black_24dp,
+                trailingIcon = {
+                    var image = if (viewConfirmPassword) Icons.Filled.Visibility
+                    else Icons.Filled.VisibilityOff
+
+                    if (!confirmPasswordError) {
+                        IconButton(onClick = { viewConfirmPassword = !viewConfirmPassword }) {
+                            Icon(
+                                imageVector = image,
+                                contentDescription = "password visibility Button"
+                            )
+                        }
+                    } else {
+                        Icon(imageVector = Icons.Filled.Error, contentDescription = null)
+                    }
+                },
+                error = confirmPasswordError,
+                visualTransformation = if (viewConfirmPassword) VisualTransformation.None else PasswordVisualTransformation(),
+                keyboardType = KeyboardType.Password
+
+            )
+
+            Button(
+                onClick = {
+                    keyboardController?.hide()
+                    newPasswordErrorContent = viewModel.getNewPasswordError(newPassword)
+                    confirmPasswordErrorContent = viewModel.getConfirmPasswordError(confirmPassword)
+
+                    when {
+                        newPasswordErrorContent.isEmpty() && confirmPasswordErrorContent.isEmpty() -> {
+                            viewModel.validateAndUpdatePassword(
+                                newPassword,
+                                confirmPassword
+                            )
+                        }
+
+                        newPasswordErrorContent.isEmpty() && confirmPasswordErrorContent.isNotEmpty() -> {
+                            confirmPasswordError = true
+                        }
+
+                        newPasswordErrorContent.isNotEmpty() && confirmPasswordErrorContent.isEmpty() -> {
+                            newPasswordError = true
+                        }
+
+                        else -> {
+                            confirmPasswordError = true
+                            newPasswordError = true
+                        }
```

Comment:
imo this logic should have been handled in the viewmodel

Reply:
but how do we update `confirmPasswordError = true`  and  `newPasswordError = true` then?